TypeSystem/flow-analysis/reachability_A22_t11.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion - Method invocation: If `N` is an expression of the form
+/// `E1.m1(E2)`, then:
+/// - Let `before(E1) = before(N)`
+/// - Let `before(E2) = after(E1)`
+/// - Let `T` be the static return type of the invocation
+/// - If `T <: Never` then:
+///   - Let `after(N) = unreachable(after(E2))`.
+/// - Otherwise:
+///   - Let `after(N) = after(E2)`.
+///
+/// @description Checks that if in the expression of the form `E1..m1(E2)` the
+/// return type of `m1(...)` is `Never` then `after(N) = unreachable(after(E2))`
+/// @author sgrekhov22@gmail.com
+
+class C<T extends Never> {
+  T foo() => throw "";
+  Future<T> bar() async {
+    throw "";
+  }
+  void qux(int _) {}
+}
+
+void test1() {
+  late int i;
+  if (2 > 1) {
+    C()
+      ..foo()
+      ..qux(i = 42);
+  }
+  i; // Definitely unassigned
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+void test2() {
+  late int i;
+  if (2 > 1) {
+    C()
+      ..bar() // Return type is not `Never`
+      ..qux(i = 42);
+  }
+  i; // Not definitely unassigned
+}
+
+void test3() async {
+  late int i;
+  if (2 > 1) {
+    await C()
```

Comment:
Please simplify this a bit, or add a comment explaining what's going on. Also, I think there's a need to have `.ignore()` or we will have an exception after `main`.

Reply:
Added explanation caooment